File: src/main/java/net/allape/bus/HistoryTopicHandler.java
package net.allape.bus;

import com.intellij.util.messages.Topic;

public interface HistoryTopicHandler {

    Topic<HistoryTopicHandler> HISTORY_TOPIC = Topic.create("history topic", HistoryTopicHandler.class);

    void before (HAction action);

    void after (HAction action);

    enum HAction {
        // 重新渲染传输历史窗口的内容
        RERENDER,
    }

}

File: src/main/kotlin/net/allape/common/HistoryTopicHandler.kt
package net.allape.common

import com.intellij.util.messages.Topic

interface HistoryTopicHandler {

    companion object {

        var HISTORY_TOPIC = Topic.create("history topic", HistoryTopicHandler::class.java)

    }

    fun before(action: HAction?)

    fun after(action: HAction?)

    enum class HAction {
        // 重新渲染传输历史窗口的内容
        RERENDER
    }
}